distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -0,0 +1,194 @@
+import project.Any.Any
+import project.Data.Array.Array
+import project.Data.Json.Json
+import project.Data.Locale.Locale
+import project.Data.Map.Map
+import project.Data.Numbers.Number
+import project.Data.Range.Extensions
+import project.Data.Text.Text
+import project.Data.Vector.Vector
+import project.Error.Error
+import project.Meta
+import project.Nothing.Nothing
+
+from project.Data.Boolean import Boolean, True, False
+
+## ALIAS To JSON
+
+   Generically converts an atom into a JSON object.
+
+   The input atom is converted into a JSON object, with a `"type"` field set to
+   the atom's type name and all other fields serialized with their name as
+   object key and the value as the object value.
+
+   > Example
+     Convert a vector to JSON.
+         [1, 2, 3, 4].to_json
+Any.to_json : Json
+Any.to_json self =
+    m = Meta.meta self
+    case m of
+        _ : Meta.Atom ->
+            cons = Meta.Constructor.Constructor_Data m.constructor
+            fs = m.fields
+            fnames = cons.fields
+            json_fs = 0.up_to fnames.length . fold Map.empty m-> i->
+                m.insert (fnames.at i) (fs.at i . to_json)
+            with_tp = json_fs . insert "type" (Json.String cons.name)
+            Json.Object with_tp
+        _ : Meta.Constructor ->
+            Json.Object (Map.empty . insert "type" (Json.String m.name))
+
+        ## The following two cases cannot be handled generically and should
+           instead define their own `to_json` implementations.
+        _ : Meta.Polyglot -> Json.Null
+        _ : Meta.Primitive -> Json.Null
+
+## UNSTABLE
+   ADVANCED
+
+   Returns a Text used to display this value in the IDE.
+
+   The particular representation is left unspecified and subject to change in
+   the future. The current implementation uses JSON serialization as the
+   default.
+
+   Types defining their own versions of this method should ensure that the
+   result is reasonably small and that the operation is quick to compute.
+
+   > Example
+     Converting the number `2` into visualization data.
+
+         2.to_default_visualization_data
+Any.to_default_visualization_data : Text
```

Comment:
Shouldn't the visualization extension method be defined in `Standard.Visualization` library?

Reply:
Yes that's a good plan - I've got a little decoupling to do in the next PR (removing Geo from Table) so I'll move these then.

Reply:
+1, but ok if done later